### ISSUE TITLE
simple: spot unnecessary nil checks around a range

### ISSUE THREAD

**Issue by mvdan** [dominikh/go-tools#67]
```
if strs != nil {
        for _, s := range strs {
                println(s)
        }
}
```

can be

```
for _, s := range strs {
        println(s)
}
```

This is a common misconception among people coming from lanugages like Java, where iterating over a nil list is a crash and not zero work.

**Comment by mvdan** [comment]
I just realised the same case applies for maps. And even when reading a single element:

```
if m != nil {
    v := m[k]
    [...]
}
```

The nil check is redundant there too.

**Comment by dominikh** [comment]
Implemented the check for check + loop in #148. 

Will not address the map indexing case for now. We'd have to make sure that the map isn't also being inserted into, and I'm not yet convinced nil check + map index is seen in the wild. Feel free to open a separate issue if this is a code smell existing in the real world.
